File: code/section_iii/3_5_update_and_delete.py
from sqlalchemy import create_engine, text

engine = create_engine('sqlite:///thunderbird_manufacturing.db')
conn = engine.connect()

# DELETE records
deleteStmt = text("DELETE FROM CUSTOMER WHERE CUSTOMER_ID > :id")
conn.execute(deleteStmt, id=10)
print("Records deleted")

# UPDATE records
updateStmt = text("UPDATE CUSTOMER SET STATE = lower(STATE)")
conn.execute(updateStmt)
print("Records updated, STATE made lowercase")

# SELECT records to check them 
for r in conn.execute(text("SELECT * FROM CUSTOMER")):
    print(r)

# UPDATE records, make STATE UPPERCASE again
updateStmt = text("UPDATE CUSTOMER SET STATE = upper(STATE)")
conn.execute(updateStmt)


conn.close()

File: code/section_iii/3_5_update_and_delete.java
import java.sql.*;

public class JavaLauncher {

    public static void main(String[] args) {

        try {

            Connection conn = DriverManager.getConnection("jdbc:sqlite:/c:/git/oreilly_advanced_sql_for_data/thunderbird_manufacturing.db");

            // Delete customers with ID greater than 10
            PreparedStatement ps = conn.prepareStatement("DELETE FROM CUSTOMER WHERE CUSTOMER_ID > ?");
            ps.setInt(1,10);

            ps.execute();


            // Update customers
            ps = conn.prepareStatement("UPDATE CUSTOMER SET STATE = lower(STATE)");
            ps.execute();

            // Iterate customers
            ResultSet rs = conn.prepareStatement("SELECT * FROM CUSTOMER").executeQuery();

            while (rs.next()) {
                System.out.println(rs.getString("CUSTOMER_ID") + " " + rs.getString("CUSTOMER_NAME") + " " + rs.getString("STATE"));
            }
            //release connection
            conn.close();

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}